panel-orig.js

```
@@ -0,0 +1,444 @@
+        function (result, error) {
+          if (error) {
+            throw error;
+          }
+          if (event.detail.reEval) {
+            // The property requires a re-eval because it is accessor
+            // and O.o() won't update it.
+            refreshProperty(key, childTree, path, isModel);
+          }
+        }
+      );
+    });
+    window.addEventListener('refresh-property', function (event) {
+      var key = elementTree.selectedChild.key;
+      var childTree = event.detail.tree;
+      var path = event.detail.path;
+      var isModel = (event.target.id === 'model-tree');
+      refreshProperty(key, childTree, path, isModel);
+    });
+    window.addEventListener('object-expand', function (event) {
+      var isModel = (event.target.id === 'model-tree');
+      expandObject(event.detail.path, isModel);
+    });
+    // An object has been collapsed. We must remove the object observer
+    // and empty the index-propName map in the host page for this object
+    window.addEventListener('object-collapse', function (event) {
+      var key = elementTree.selectedChild.key;
+      var path = event.detail.path;
+      var isModel = (event.target.id === 'model-tree');
+      EvalHelper.executeFunction('removeObjectObserver', [key, path, isModel], function (result, error) {
+        if (error) {
+          throw error;
+        }
+        EvalHelper.executeFunction('emptyIndexMap', [key, path, isModel], function (result, error) {
+          if (error) {
+            throw error;
+          }
+        });
+      });
+    });
+    window.addEventListener('breakpoint-toggle', function (event) {
+      var key = elementTree.selectedChild.key;
+      var index = event.detail.index;
+      var functionName = event.detail.isSet ? 'setBreakpoint' : 'clearBreakpoint';
+      EvalHelper.executeFunction(functionName, [key, [index]], function (result, error) {
+        if (error) {
```

At this stage shall leave them like this? I'll be able to quickly detect bugs this way. I wouldn't want anything silently failing right now.